### DISCUSSION TITLE
Usage in general (new to grafane etc)

### DISCUSSION THREAD

**Discussion by alturismo** [grafana/grafana-json-datasource#62] [Q&A]
May as a starter for me to get some sample usage here, i setted up a REST serving GPU data in json formatting which i would like to bring to my grafana dash

respond as sample from http://192.168.1.210:55555

`[
  {
    "SensorApp": "GPU-Z",
    "SensorClass": "GPU",
    "SensorName": "GPU Clock",
    "SensorValue": "300",
    "SensorUnit": "MHz",
    "SensorUpdateTime": 264958
  },
  {
    "SensorApp": "GPU-Z",
    "SensorClass": "GPU",
    "SensorName": "Memory Clock",
    "SensorValue": "101,25",
    "SensorUnit": "MHz",
    "SensorUpdateTime": 264958
  },
  {
    "SensorApp": "GPU-Z",
    "SensorClass": "GPU",
    "SensorName": "GPU Temperature",
    "SensorValue": "34",
    "SensorUnit": "Â°C",
    "SensorUpdateTime": 264958
  },
...
  {
    "SensorApp": "GPU-Z",
    "SensorClass": "GPU",
    "SensorName": "PerfCap Reason",
    "SensorValue": "1",
    "SensorUnit": "",
    "SensorUpdateTime": 264958
  },
  {
    "SensorApp": "GPU-Z",
    "SensorClass": "GPU",
    "SensorName": "GPU Voltage",
    "SensorValue": "0,725",
    "SensorUnit": "V",
    "SensorUpdateTime": 264958
  }
]`

![image](https://user-images.githubusercontent.com/8406490/107874319-0497e280-6eb9-11eb-9730-9ff8634ece52.png)

now, i added the datasource standard via url 
![image](https://user-images.githubusercontent.com/8406490/107874261-89363100-6eb8-11eb-80b1-6e43c47cc949.png)

now comes the interesting part, i guess its not plug and play so i have to add queries

![image](https://user-images.githubusercontent.com/8406490/107874282-bedb1a00-6eb8-11eb-99f5-bb01215c192b.png)

may a startpoint how to use it to get the values properly readed ?

for any Tipps, thanks ahead

**Comment by alturismo** [comment]
ok, 1 step further ;)

![image](https://user-images.githubusercontent.com/8406490/107879475-48024900-6ed9-11eb-9e77-224b49f49b14.png)

now i take a look how to make them as graphs with timeline, any hints, thanks ahead ;)

**Reply by marcusolsson** [reply]
Looking good! 

Each visualisation in Grafana requires queries with specific sets of field types. For example, the Graph panel can graph any query that return a Time and a Number field.

**Comment by alturismo** [comment]
thanks, i d need to know how to sep. fetch then individual terms from the json i assume, would be nice for a sample like "CPU Clock" in case u have the time.

**Reply by marcusolsson** [reply]
What do you mean by "individual terms"? The JSON Path can return single values, and not just arrays?

**Reply by alturismo** [reply]
im sure they do, i just dont know how (yet) ;)

**Reply by marcusolsson** [reply]
Would something like this work for you?

<img width="734" alt="CleanShot 2021-02-16 at 19 26 42@2x" src="https://user-images.githubusercontent.com/8396880/108105460-f00c4380-708c-11eb-940a-dbb770112107.png">

**JSON data:**

```
{
  "host": "instance-1",
  "cpu": 0.45
}
```

**Reply by alturismo** [reply]
thanks for trying to help, i spent some time now but actually with no result to get a scaled graph.

i stay simple now with the table i have ;)

goal was more as sample some graphical showoffs

![image](https://user-images.githubusercontent.com/8406490/108552353-79bc4b00-72f1-11eb-955b-e6efb03a25f9.png)

thanks again for pointing, but somehow i dont understand the query's and how to use them properly, nevermind now ;)
